frontend/talentsearch/resources/js/components/skills/AddSkillsToExperience/AddSkillsToExperience.tsx

```diff
@@ -0,0 +1,211 @@
+  onAddSkill: (id: Scalars["ID"]) => Promise<void>;
+}
+
+const AddSkillsToExperience: React.FunctionComponent<
+  AddSkillsToExperienceProps
+> = ({ allSkills, frequentSkills, addedSkills, onRemoveSkill, onAddSkill }) => {
+  const intl = useIntl();
+  const locale = getLocale(intl);
+
+  const [familyFilteredSkills, setFamilyFilteredSkills] = useState<Skill[]>([]);
+  const [searchFilteredSkills, setSearchFilteredSkills] = useState<Skill[]>([]);
+
+  const resultsPaginationPageSize = 5;
+  const frequentSkillsPagination = usePaginationVars<Skill>(
+    1,
+    resultsPaginationPageSize,
+    frequentSkills,
+  );
+  const mainstreamSkillsPagination = usePaginationVars<Skill>(
+    1,
+    resultsPaginationPageSize,
+    familyFilteredSkills,
+  );
+  const keywordSearchPagination = usePaginationVars<Skill>(
+    1,
+    resultsPaginationPageSize,
+    searchFilteredSkills,
+  );
+
+  /**
+   * A handler which takes a list of skill families and filters the allSkills list to
+   * any skills that are a part of those families.  Applies the filter
+   * internally using the useState hook.
+   * @param {SkillFamily[]} checkedFamilies - The selected skill families to filter against.
+   */
+  const handleSkillFamilyChange = (checkedFamilies: SkillFamily[]): void => {
+    const checkedFamilyIds = checkedFamilies.map((family) => family.id);
+    const matchingSkills = allSkills.filter((skill) =>
+      // https://stackoverflow.com/a/39893636
+      skill.families?.some((skillFamily) =>
+        checkedFamilyIds.includes(skillFamily.id),
+      ),
+    );
+    setFamilyFilteredSkills(matchingSkills);
+    mainstreamSkillsPagination.setCurrentPage(1);
+  };
+
+  /**
+   * A handler which takes a search query and uses a matching function to filter
+   * a list of skills.  Applies the filter internally using the useState hook.
+   * @param {string} searchQuery - The search text to filter against.
+   */
+  const handleSearch = (searchQuery: string): Promise<void> => {
+    return new Promise<void>((resolve) => {
+      const matchedSkills = allSkills.filter((skill) =>
+        matchStringCaseDiacriticInsensitive(
+          searchQuery,
+          skill.name[locale] ?? "",
+        ),
+      );
+      setSearchFilteredSkills(matchedSkills);
+      keywordSearchPagination.setCurrentPage(1); // just in case the new list of matched skills requires fewer pages
+      resolve();
+    });
+  };
+
+  // this function can be a bit heavy
+  const allSkillFamilies = useMemo(
+    () => invertSkillTree(allSkills),
+    [allSkills],
+  );
+
+  return (
+    <>
+      <AddedSkills skills={addedSkills} onRemoveSkill={onRemoveSkill} />
+      <hr />
+      <h5>
+        {intl.formatMessage({
+          defaultMessage: "Add Skills",
+          description: "Section header for adding skills to this experience",
+        })}
+      </h5>
+      <TabSet>
+        <Tab
+          icon={<SearchCircleIcon style={{ width: "1rem" }} />}
+          text="My frequent skills"
```

Comment:
I believe this should be also be translatable

Reply:
Yep, good catch!